File: backend/PM.Web/Persons/DataModels/Person.cs
using PM.Web.Data.Entities;
using PM.Web.Persons.Enums;
using System;
using System.ComponentModel.DataAnnotations;

namespace PM.Web.Persons.DataModels
{
    public class Person : IEntityWithId
    {
        public int Id { get; set; }

        [MaxLength(25)]
        public string FirstName { get; set; }

        [MaxLength(50)]
        public string LastName { get; set; }

        [MaxLength(11)]
        public string PersonalNumber { get; set; }

        public DateTime BirthDate { get; set; }

        public Gender Gender { get; set; }

        public decimal Salary { get; set; }
    }
}

File: frontend/src/app/models/person.ts
import { Gender } from "../components/persons/gender";

export interface Person {
  id: number;
  firstName: string;
  lastName: string;
  personalNumber: string;
  birthDate
  gender: Gender;
  salary: number;
}

export const personProps = {
  id: "id",
  firstName: "firstName",
  lastName: "lastName",
  personalNumber: "personalNumber",
  birthDate: "birthDate",
  gender: "gender",
  salary: "salary",
};
